File: TareaGo/Kata19/Palabras.go
package main 

import (
	"fmt"
	"os"
)

func leerArchivo() string {
	contenido, _ := os.Open("palabras.txt")

	contenido_bytes := make([]byte, 794505)
	contenido.Read(contenido_bytes)
	cadena := string(contenido_bytes)
	contenido.Close()
	
	return cadena
}

func convertirArreglo() []string {
	cadena := leerArchivo()

	palabras := make([]string, 84099)
	var cad string
	idx := 0

	for i := 0; i < 794504; i++ {
		if cadena[i] != '\n' {
			cad += string(cadena[i])
		} else {
			palabras[idx] = cad
			idx++
			cad = ""
		}
	}

	return palabras
}

func imprimirArreglo(palabras []string) {
	for i := 0; i < len(palabras); i++ {
		fmt.Println(palabras[i])
	}
}

func seleccionarSimilares(palabra string, palabras []string) []string {
	
	similares := make([]string, 0)

	for i := 0; i < len(palabras); i++ {
		if len(palabra) == len(palabras[i]) {
			similares = append(similares, palabras[i])
		}
	}
	return similares
}

func delimitarPalabra(entrada string, salida string, similares []string) []string {
	
	palabras := make([]string, 0)

	pos1 := buscarPalabra(entrada, similares)
	pos2 := buscarPalabra(salida, similares)

	if pos1 != -1 || pos2 != -1 {
		if pos1 < pos2 {
			for i := pos1; i <= pos2; i++ {
				palabras = append(palabras, similares[i])
			}	
		}
		if pos1 > pos2 {
			for i := pos1; i >= pos2; i-- {
				palabras = append(palabras, similares[i])
			}
		}
	}
	return palabras
}

func buscarPalabra(palabra string, similares []string) int {

	for i := 0; i < len(similares); i++ {
		if similares[i] == palabra {
			return i
		}
	}
	return -1
}


func main() {
	
	palabras := convertirArreglo()
	similares := seleccionarSimilares("lead", palabras)
	palabras_delimitadas := delimitarPalabra("lead", "gold", similares)

	imprimirArreglo(palabras_delimitadas)
}

File: TareaPython/Kata19/Palabras.py

class Palabra(object):
	
	def __init__(self):
		self.arreglo = []
		self.arregloLongi = [] #para palabras de la misma longitud
		self.palabrasDelimitadas = [] #para acotar las posibilidades

	def agregarPalabra(self, palabra):
		self.arreglo.append(palabra)

	def agregarPalabraML(self, palabraOrigen, palabraDestino):
		if len(palabraOrigen) == len(palabraDestino):
			for x in range(len(self.arreglo)):
				if len(self.arreglo[x])-1 == len(palabraOrigen):
					self.arregloLongi.append(self.arreglo[x])
			self.delimitarPalabra(palabraOrigen, palabraDestino)

	def buscarPalabra(self, palabra):
		for x in range(len(self.arregloLongi)):
			if self.arregloLongi[x][0] == palabra[0]:
				return x
		return -1


	def delimitarPalabra(self, palabraOrigen, palabraDestino):
		pos1 = self.buscarPalabra(palabraOrigen)
		pos2 = self.buscarPalabra(palabraDestino)

		if pos1 != -1 or pos2 != -1:
			if pos1 < pos2:
				while pos1 <= pos2:
					self.palabrasDelimitadas.append(self.arregloLongi[pos1])
					pos1+=1

			if pos1 > pos2:
				while pos1 >= pos2:
					self.palabrasDelimitadas.append(self.arregloLongi[pos1])
					pos1-=1

	def leerArchivo(self):
		archivo = open("palabras.txt")
		
		for linea in archivo.readlines():
			self.agregarPalabra(linea)
		archivo.close()

	def imprimir(self, arreglo):
		for x in range(len(arreglo)):
			print arreglo[x],
		
def main():
	palabra = Palabra()

	palabra.leerArchivo()
	palabra.agregarPalabraML('zero', 'soap')
	palabra.imprimir(palabra.palabrasDelimitadas)

if __name__ == '__main__':
	main()